packages/purpose-platformstate-writer/src/consumerServiceV1.ts

```diff
@@ -1,19 +1,137 @@
 import { match } from "ts-pattern";
-import { PurposeEventEnvelopeV1 } from "pagopa-interop-models";
+import {
+  fromPurposeV1,
+  makePlatformStatesPurposePK,
+  missingKafkaMessageDataError,
+  PlatformStatesPurposeEntry,
+  Purpose,
+  PurposeEventEnvelopeV1,
+  PurposeV1,
+  purposeVersionState,
+} from "pagopa-interop-models";
 import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
+import {
+  deletePlatformPurposeEntry,
+  getPurposeStateFromPurposeVersions,
+  updatePurposeDataInTokenGenerationStatesTable,
+  readPlatformPurposeEntry,
+  updatePurposeDataInPlatformStatesEntry,
+  writePlatformPurposeEntry,
+  updatePurposeEntriesInTokenGenerationStatesTable,
+} from "./utils.js";
 
 export async function handleMessageV1(
   message: PurposeEventEnvelopeV1,
-  _dynamoDBClient: DynamoDBClient
+  dynamoDBClient: DynamoDBClient
 ): Promise<void> {
   await match(message)
+    .with({ type: "PurposeVersionActivated" }, async (msg) => {
+      const purpose = parsePurpose(msg.data.purpose, msg.type);
+      const purposeState = getPurposeStateFromPurposeVersions(purpose.versions);
+      const primaryKey = makePlatformStatesPurposePK(purpose.id);
+      const existingPurposeEntry = await readPlatformPurposeEntry(
+        dynamoDBClient,
+        primaryKey
+      );
+      const purposeVersion = purpose.versions
+        .slice()
```

Comment:
same considerations [here](https://github.com/pagopa/interop-be-monorepo/pull/972#discussion_r1796905886)

Reply:
Updated to use the function instead